### DISCUSSION TITLE
I want to add a feature myself and eventually add a PR if it's useful. Not sure how to start though.

### DISCUSSION THREAD

**Discussion by tracure1337** [will-stone/browserosaurus#521] [Ideas]
Hi,

Thanks for this awesome product once again :-)

So I'd like to populate a sqlite db with a list of urls from all my obsidian notebooks and then basically create a small popup (or even direct link to entry) IF the URL I am trying to visit already exists in my notes.

Are there any short pointers where in the code I should start reading or some DO NOTS that I should be aware of?

Thanks for input!

**Comment by will-stone** [comment]
Hi @cocoonkid,

Are you talking about adding rules to Browserosaurus? If so, that's out of scope for this project, but you can combine Browserosaurus with John Sterling's [Finicky](https://github.com/johnste/finicky) to achieve what you need.

Thanks,

Will.

**Reply by tracure1337** [reply]
Thanks for the swift reply.

Finicky will help with opening obsidian at the correct place UPON Browserosaurus NOTIFYING that URL exists already.

I'd like to fork (probably only used by me) and add this rule set myself.

So I am asking where should I start and what to be aware of  as you probably already thought about this once.

Thanks @will-stone

**Reply by will-stone** [reply]
Could you set Finicky as the default browser, then have it read your db of links and act appropriately, then fallback to Browserosaurus for all _other_ links?

**Reply by tracure1337** [reply]
Thanks. I will work through boths code and see what will be easier.  Thanks again :-)
